templates/base/head_script.tmpl

```diff
@@ -9,7 +9,7 @@ If you introduce mistakes in it, Gitea JavaScript code wouldn't run correctly.
 		appVer: '{{AppVer}}',
 		appUrl: '{{AppUrl}}',
 		appSubUrl: '{{AppSubUrl}}',
-		assetVersion: '{{AssetVersion}}',
+		assetVersionEncoded: encodeURIComponent('{{AssetVersion}}'), // will be used in URL construction directly
```

Comment:
Why need to encode again here? Shouldn't `html/template` escape it, or is the contextual escaping failing in such a case?

Reply:
Here is a JS context, `html/template` will only encode the AssetVersion into a JS string.

Reply:
I see. [Go's autoescaping](https://go.dev/src/html/template/escape.go) is really an insane way of doing escaping, as it forces the user to need to know when it will escape. A much more sane way would be how original Handlebars did it, e.g. escape everything and let user opt out via triple quotes, e.g. `{{{value}}}`.